### ISSUE TITLE
Remove shims of inactive installed versions

### ISSUE THREAD

**Issue by michi-zuri** [asdf-community/asdf-direnv#10]
When installing miniconda in asdf, plenty of shims are added that shadow system tools, including idle, clear etc. I may have the python plugin installed with Python 3.8.0 for global development and miniconda for a specific project. The additional shims added by miniconda should not shadow anything if I'm not in the specific project folder.

Wouldn't it make senseto remove the default shims folder from PATH by direnv? Or would it cause unintended side effects? Or is it maybe only a problem with just a small number of tools like miniconda/anaconda that should be avoided in asdf?

**Comment by michi-zuri** [comment]
I currently have a working asdf environment with the following languages/packages installed:

```
direnv
  2.20.0
elixir
  1.9.4-otp-22
elm
  0.19.0
  0.19.1
julia
  1.2.0
python
  miniconda3-latest
  pypy3.6-7.2.0
```

The changes I made to my configuration to get this working without conflicts were:
In .asdf/asdf.sh I commented out line 31: `#PATH="${ASDF_USER_SHIMS}:$PATH"` to avoid the default shims to be added to my PATH. 

Then I added symlinks to direnv and direnv_user_asdf in a folder that is in my path, because otherwise direnv obviously doesn't work:
```
ln -s ~/.asdf/plugins/direnv/shims/direnv_use_asdf ~/.direnvbin/direnv_use_asdf
ln -s ~/.asdf/shims ~/.direnvbin/direnv
```

My .bash_profile looks like this:
```
### ERLANG
# Add Erlang installer (from erlang-solutions) active version to path
[[ $PATH == *".erlangInstaller/default/bin"*    ]] || export PATH="~/.erlangInstaller/default/bin:$PATH"

### ASDF
# asdf shim hooks and autocompletion
. ~/.asdf/asdf.sh #with line 31 commented out to prevent inclusion of default shims
. ~/.asdf/completions/asdf.bash

### ASDF-DIRENV
# direnv executables added to the end of path, needed for initialization.
[[ $PATH == *".direnvbin"*    ]] || export PATH="$PATH:~/.direnvbin"
# direnv magic hook ;
eval "$(~/.asdf/shims/direnv hook bash)"
```

**Comment by vic** [comment]
Certainly possible, as you just showed.

I guess you could get rid of asdf `bin` and `shims` altogether. 

What if you try something like:

```bash
# Dont source ~/.asdf/asdf.sh here, just export the asdf location
export ASDF_DIR=$HOME/.asdf
eval "$(~/.asdf/shims/direnv hook bash)"
```

Then in your `.envrc` file you can have
```bash
source $ASDF_DIR/plugins/direnv/shims/direnv_use_asdf
use asdf
```

The `direnv_use_asdf` sources `$ASDF_DIR/lib/utils.sh` but doesn't actually invokes asdf as a command. So this might work, could you give it a try?

**Comment by vic** [comment]
And maybe with a small change at 

`https://github.com/asdf-community/asdf-direnv/blob/1e1082352102be29ba7a51c1bf048a440f42decd/shims/direnv_use_asdf#L73`

`_xargs_par direnv_use_asdf` -> `_xargs_par $0` in order to not depend on `direnv_use_asdf` being on PATH.

**Comment by michi-zuri** [comment]
Great appraoch, but that didn't quite work yet, probably for some obscure syntax reason:
With `$0` I get `/bin/bash: do: No such file or directory` once for every line in .tool-versions
and when I add `echo "This file is called: $0"` just above line 73, I get this additional output once: `This file is called: /bin/bash`

Reading up on the subject of `$0` I found that [$BASH_SOURCE should be more reliable.](https://mywiki.wooledge.org/BashFAQ/028)

The solution is to fetch the file path before calling watch_file:
```
_load_plugins_env() {
  local tool_versions=$1
  _this_file="$BASH_SOURCE" #needs to come before watch_file                                  
  watch_file "$tool_versions"
  eval "$(_plugins_in_file "$tool_versions" | _xargs_par $_this_file 'do' _plugin_env_bash)"
}
```
I used the changes you suggested to my .bash_profile, with the addition of `.asdf/bin` to PATH, because asdf is not handled by direnv and I still want to be able to use that command.
My .bash_profile now looks like this:
```
#... other stuff, then:
### asdf and direnv initialisation
# First adding asdf to (the end of) PATH: no need to shadow anything
# Not sourcing asdf.sh here! I don't want any shims in PATH
[[ $PATH == *"asdf/bin"* ]] || export PATH="$PATH:~/.asdf/bin"
# Secondly, adding completions to asdf command
. ~/.asdf/completions/asdf.bash
# Finally, hooking direnv plugin into bash
eval "$(asdf exec direnv hook bash)"
# done.
``` 
and my .envrc looks like this:
```
source $(asdf which direnv_use_asdf)
use asdf
```
It works beautifully! I'll try to do a fork with these changes and make a pull request. I'll also try to find some issues in various plugin issue trackers that would be solved by this approach.

**Comment by michi-zuri** [comment]
Some related issues, there are probably more:
https://github.com/danhper/asdf-python/issues/54
https://github.com/asdf-vm/asdf/issues/505
https://github.com/danhper/asdf-python/issues/44

**Comment by vic** [comment]
Awesome, @michi-zuri, thanks for the PR! Looking at it.

**Comment by vic** [comment]
Fixed in #11

**Comment by noahgaertner** [comment]
Did this change get reverted at some point? I've been using asdf-direnv for over a year now, and it's been great, but I just installed mambaforge (basically miniconda/miniforge) alongside my previous python 3.10.4 install, and whenever I type clear in zsh without the active python version being mambaforge, I get an error saying mambaforge needs to be my active python version. To my understanding, this commit should have fixed that issue, so do I need to do something else to make it work or is it just no longer supported?
